### PR TITLE
Feature/notify user final decision on request

### PR DESCRIPTION
**Pull request by oijusti**
- Checks approval status
- Updates approval status in the DB
- Notifies approval response
- Ignores approval action if acronym already was approved

Screen Shot 2021-07-22 at 8 03 07 AM

### CODE REVIEW

File: lambda/add_meaning.py [hunk 1]
```diff
@@ -177,6 +184,70 @@ def create_approval_request(acronym, definition, meaning, team_domain, user_id,
...
+            return
+
+    notify_approval_response(acronym,approved,requester_id)
+
+
+def notify_approval_response(acronym, approved, requester_id):
+    print("Sending approval response...")
+
+    if approved == True:
+        message = "Your submission for *" + acronym + "* was approved. Thanks for contributing!"
```
Comment (ne-cabrerav) [inline lambda/add_meaning.py:224]: You could use f-strings to format this string. For example: `f"Your submission for *{acronym}* was approved. Thanks for contributing!"`

File: lambda/add_meaning.py [hunk 2]
```diff
@@ -177,6 +184,70 @@ def create_approval_request(acronym, definition, meaning, team_domain, user_id,
...
+    else:
+        message = "Sorry, your submission for *" + acronym + "* was denied."
```
Comment (ne-cabrerav) [inline lambda/add_meaning.py:226]: Same here

File: lambda/add_meaning.py [hunk 3]
```diff
@@ -177,6 +184,70 @@ def create_approval_request(acronym, definition, meaning, team_domain, user_id,
         else:
             print("Skip approver due to approver sent acronym request")
 
+
+def check_approval_status(acronym):
```
Comment (ne-cabrerav) [inline lambda/add_meaning.py:188]: I think this method is not necessary because we can know the approval status from the `persistDecision` method and from there we can call the `notify_approval_response` method. You could create a method to update the approval status and also call it from `persistDecision` accordingly

File: lambda/add_meaning.py [hunk 4]
```diff
@@ -177,6 +184,70 @@ def create_approval_request(acronym, definition, meaning, team_domain, user_id,
...
+            ExpressionAttributeValues={
+                ':d': APPROVAL_STATUS_APPROVED,
+            },
+            ReturnValues="UPDATED_NEW"
+        )
+        print("response: " + str(response))
+    else:
+        if len(deniers) >= REVIEWERS_MAX:
+            approved = False
+            # TODO: Remove item from DB
```
Comment (ne-cabrerav) [inline lambda/add_meaning.py:213]: The removal from the DB is done in another PR in the `persistDecision` method

File: lambda/add_meaning.py [hunk 5]
```diff
@@ -249,8 +320,10 @@ def persistDecision(acronym, userId, decision):
     
     decisionStr = APPROVERS_STR if decision else DENIERS_STR
     reviewers = result['Items'][0].get(decisionStr, [])
+    approval_status = result['Items'][0].get(APPROVAL_STR)
 
-    if checkAlreadyReviewed(result, userId):
+    # TODO: Ignore approval action
+    if checkAlreadyReviewed(result, userId) or approval_status == APPROVAL_STATUS_APPROVED:
```
Comment (ne-cabrerav) [inline lambda/add_meaning.py:326]: We could only check for the `approval_status` here

**Review by ne-cabrerav — CHANGES_REQUESTED** [review]

File: lambda/add_meaning.py (hunk 1 above)
Comment (oijusti) [reply lambda/add_meaning.py]: Thanks, done!

File: lambda/add_meaning.py (hunk 2 above)
Comment (oijusti) [reply lambda/add_meaning.py]: Thanks, done!

File: lambda/add_meaning.py [hunk 6]
```diff
@@ -44,14 +49,16 @@ def explain(acronym):
...
     
     results = table.put_item(
         Item={
             'Acronym': acronym,
             'Definition': definition,
             'Meaning': meaning,
-            'Notes': notes
+            'Notes': notes,
+            REQUESTER_STR: user_id,
+            APPROVAL_STR: APPROVAL_STATUS_PENDING
```
Comment (gihernandez) [inline lambda/add_meaning.py:61]: This would better be **APPROVAL_STATUS**

**Review by gihernandez — APPROVED** [review]

File: lambda/add_meaning.py (hunk 3 above)
Comment (oijusti) [reply lambda/add_meaning.py]: Thanks, done!

File: lambda/add_meaning.py (hunk 4 above)
Comment (oijusti) [reply lambda/add_meaning.py]: Got it

File: lambda/add_meaning.py (hunk 5 above)
Comment (oijusti) [reply lambda/add_meaning.py:326]: The **checkAlreadyReviewed** method should not be needed after the task **Update approver slack message form** is complete.
Here, instead of returning error 400 we still need to update the approver slack form properly to avoid confusion.

File: lambda/add_meaning.py (hunk 6 above)
Comment (oijusti) [reply lambda/add_meaning.py:61]: I considered using **APPROVAL_STATUS**, but I think it is a bit redundant, that's why I kept it simple with one word.
